File: concepts/functions/calculator.py
""" Highest order function to calculate two numbers given an operator """


def calculator(num1, num2, operator):
    return operator(num1, num2)


def add(num1, num2):
    return num1 + num2


def subtract(num1, num2):
    return num1 - num2


def multiplied_by(num1, num2):
    return num1 * num2


def divided_by(num1, num2):
    return num1 / num2


def raised_to_power_of(num1, num2):
    return num1 ** num2


def quotient(num1, num2):
    return num1 // num2


def modulus(num1, num2):
    return num1 % num2


# Create an array to contain each of the operators
operators = [add, subtract, multiplied_by, divided_by,
    raised_to_power_of, quotient, modulus]

num1 = 100; num2 = 8
for operator in operators:
    answer = calculator(num1, num2, operator)
    print(f'{num1} {operator.__name__} {num2} = {answer}')

File: concepts/functions/calculator.js
/* Highest order function to calculate two numbers given an operator */


function calculator(num1, num2, operator) {
    return operator(num1, num2);
}

function add(num1, num2) {
    return num1 + num2;
}

function subtract(num1, num2) {
    return num1 - num2;
}

function multiply(num1, num2) {
    return num1 * num2;
}

function divide(num1, num2) {
    return num1 / num2;
}

function raiseToPower(num1, num2) {
    return num1 ** num2;
}

function modulo(num1, num2) {
    return num1 % num2;
}

/* Create an array to contain each of the operators */
operators = [add, subtract, multiply, divide, raiseToPower, modulo]

/* Loop over each of the operators to invoke the functions on the operands
used as arguments  */
for (var i = 0; i < operators.length; i++) {
    answer = calculator(6, 2, operators[i])
    console.log(answer)
}
